smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CodegenVisitor.java

```diff
@@ -281,4 +305,43 @@ public Void serviceShape(ServiceShape shape) {
 
         return null;
     }
+
+    private boolean shapesMatch(Shape shape, Shape otherShape) {
+        // Check names match.
+        if (!shape.getId().getName().equals(otherShape.getId().getName())) {
+            return false;
+        }
+
+        // Check traits match.
+        Map<ShapeId, Trait> traits = shape.getAllTraits();
```

Comment:
This comparison looks like it could just be: `shape.getAllTraits().equals(otherShape.getAllTraits())`.

Reply:
Fixed in 66e0ca0